src/wasm/wasm-emscripten.cpp

```diff
@@ -1224,25 +1224,31 @@ std::string EmscriptenGlueGenerator::generateEmscriptenMetadata(
   });
   meta << "\n  ],\n";
 
-  auto mainReadsParams = false;
-  auto* exp = wasm.getExportOrNull("main");
-  if (!exp) {
-    exp = wasm.getExportOrNull("__main_argc_argv");
-  }
-  if (exp) {
-    if (exp->kind == ExternalKind::Function) {
-      auto* main = wasm.getFunction(exp->value);
-      mainReadsParams = true;
-      // If main does not read its parameters, it will just be a stub that
-      // calls __original_main (which has no parameters).
-      if (auto* call = main->body->dynCast<Call>()) {
-        if (call->operands.empty()) {
-          mainReadsParams = false;
+  // In normal mode we look at the signature of the main function to determine
+  // if main raads argv.  In standalone mode we don't export main at all but
+  // _start, and in this case its easy to tell if program uses argv based on
+  // the syscalls it imports.
```

Comment:
which syscalls, btw? I'm curious, and also it might be worth mentioning here.

Reply:
__wasi_args_get and __wasi_args_sizes_get.   Included them in comments and PR description.